src/helpers/test/annotation-metadata-test.js

```diff
@@ -0,0 +1,203 @@
+import { documentMetadata, domainAndTitle } from '../annotation-metadata';
+
+const fakeAnnotation = (props = {}) => {
+  return {
+    document: {},
+    uri: 'http://example.com/a/page',
+    ...props,
+  };
+};
+
+describe('documentMetadata', () => {
+  context('when the annotation has a document property', () => {
+    it('returns the hostname from annotation.uri as the domain', () => {
+      const annotation = fakeAnnotation();
+      assert.equal(documentMetadata(annotation).domain, 'example.com');
+    });
+
+    context('when annotation.uri does not start with "urn"', () => {
+      it('uses annotation.uri as the uri', () => {
+        const annotation = fakeAnnotation();
+        assert.equal(
+          documentMetadata(annotation).uri,
+          'http://example.com/a/page',
+        );
+      });
+    });
+
+    context('when document.title is an available', () => {
+      it('uses the first document title as the title', () => {
+        const annotation = fakeAnnotation({
+          document: {
+            title: ['My Document', 'My Other Document'],
+          },
+        });
+
+        assert.equal(
+          documentMetadata(annotation).title,
+          annotation.document.title[0],
+        );
+      });
+    });
+
+    context('when there is no document.title', () => {
+      it('returns the domain as the title', () => {
+        const annotation = fakeAnnotation();
+        assert.equal(documentMetadata(annotation).title, 'example.com');
+      });
+    });
+
+    ['http://localhost:5000', '[not a URL]'].forEach(uri => {
+      it('returns empty domain if URL is invalid or private', () => {
+        const annotation = fakeAnnotation({ uri });
+        const { domain } = documentMetadata(annotation);
+        assert.equal(domain, '');
+      });
+    });
+  });
+
+  context('when the annotation does not have a document property', () => {
```

Comment:
`document` is a non-optional property of annotations. I'm guessing this was another vestige of less-strongly typed times. It is still the case that all _properties_ of a document, including the title, are optional.

Reply:
To be honest, I blindly copy-pasted the whole `describe` block and was happy because all tests passed 😅